### DISCUSSION TITLE
SmartEdit/Deepgrow section doesn't seem to be working

### DISCUSSION THREAD

**Discussion by lukasvdstricht** [Project-MONAI/MONAILabel#989] [Q&A] [answered]
Hi everyone

I am using MONAILabel to segment the left and right breast from MRI images of the chest.
To do that I used Slicer's _Grow from seeds_ functionality to annotate some images (35 to be exact). Next, I adapted the `deepedit.py` file to contain the labels I wanted (`left breast` and `right breast`) and set the flag for `use_pretrained_model` to `false`. 
I started training this `deepedit` model and quickly got some nice results!

The autosegmentation was good, but not perfect. I decided I wanted to correct a few autosegmented images using the deepgrow/smartedit functionality in the MONAILabel extension, as was done in a tutorial I'd seen about segmentation of the spleen.

For example, I got the automatic segmentation below and found it to be a little oversegmented. 
![smartedit1](https://user-images.githubusercontent.com/111343421/189598966-31039e4f-66bc-4d31-a5a8-061e3efd1340.png)

So I went to the SmartEdit/Deepgrow section and selected `background`. I placed a first point as can be seen below.
![smartedit2](https://user-images.githubusercontent.com/111343421/189600654-ea23ef4c-8e8a-43a0-88d9-da088e26c6b9.png)

The program seems to be loading for about 30 seconds. Then it stops loading but no changes to the segmentation happen. So I add another point (see below), but again nothing happens.
![smartedit3](https://user-images.githubusercontent.com/111343421/189603537-75f0df0d-0452-489e-aecd-b4e7ce6a9e3a.png)

Is there anything I'm doing wrong? Is the part I want to delete too small?

Does anyone know how this can be helped?

Thank you in advance!

**Comment by diazandr3s** [comment, marked as answer]
Hi @lukasvdstricht,


Happy to see these results.

Are you using the [default image size](https://github.com/Project-MONAI/MONAILabel/blob/main/sample-apps/radiology/lib/configs/deepedit.py#L80) for DeepEdit? The reason you're getting these blocky results is because the DeepEdit model is sampling the volumes to that image's size, which can be small for the actual image's size.

One way of improving the results is by increasing the image size to 256x256x128 - this will require more GPU memory, though

In addition to this, there was a recent improvement on the click interaction that was recently added to MONAI (https://github.com/Project-MONAI/MONAI/pull/4970). We'll see this change updated in the new MONAI Label version that will be published soon.

In the meantime, I'd suggest you used other tools available in Slicer to edit/correct the segmentation.

I hope this helps,

**Reply by lukasvdstricht** [reply]
Thank you @diazandr3s 

The increase in image size is definitely benificial for the segmentation results. It is far less blocky now!
I will keep an eye out for the next version of MONAI Label.

Thank you very much!
